sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4189,6 +4189,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_AVRO_ALLOW_READING_WITH_INCOMPATIBLE_SCHEMA =
+    buildConf("spark.sql.legacy.avro.allowReadingWithIncompatibleSchema")
+      .internal()
+      .doc("When types in Avro are encoded in the same format, but the type in the Avro schema " +
```

Comment:
The doc is confusing here...can you revise it to clarify what's the behavior with/without the conf enabled.

Reply:
We will need to update `sql-migration-guide.md`

Reply:
sure!